packages/flutter/lib/src/widgets/form.dart

```diff
@@ -250,8 +250,17 @@ class FormState extends State<Form> {
 
   void _register(FormFieldState<dynamic> field) {
     _fields.add(field);
-    if (widget.autovalidateMode == AutovalidateMode.onUnfocus) {
-      field._focusNode.addListener(() => _updateField(field));
+    switch (widget.autovalidateMode) {
+      case AutovalidateMode.always:
+        // If the field has already been validated or has no errors, we can skip the if statement to avoid an infinite loop.
```

Comment:
What's the meaning of "infinite loop" here? Is it possible for the same field to be registered multiple times?

Reply:
Since the build method is always triggered when interacting with a field, or in this case, on the first build, it was causing recursive calls, leading to a memory overflow.